packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1720,6 +1712,28 @@ export class ContainerRuntime
 		});
 	}
 
+	public getCreateChildSummarizerNodeFn(id: string, createParam: CreateChildSummarizerNodeParam) {
+		return (
+			summarizeInternal: SummarizeInternalFn,
+			getGCDataFn: (fullGC?: boolean) => Promise<IGarbageCollectionData>,
+		) =>
+			this.summarizerNode.createChild(
+				summarizeInternal,
+				id,
+				createParam,
+				undefined,
+				getGCDataFn,
+			);
+	}
+
+	public deleteChildSummarizerNode(id: string) {
+		return this.summarizerNode.deleteChild(id);
+	}
+
+	public makeLocallyVisible() {
+		assert(false, "should not be called");
```

Comment:
Might want to put a comment on why we should never get here, but this API needs to be here.
I do not see an easy path here, but maybe you can find some solution that does not require having this implementation here, but instead have it around wrapContext() call.